nixos/tests/installer.nix

```diff
@@ -68,160 +67,180 @@ let
+      def create_machine_from_file(machine_config_file):
+          with open(machine_config_file, "r") as f:
+              return create_machine(json.load(f))
+
 
-      $machine->start;
+      machine.start()
 
       # Make sure that we get a login prompt etc.
-      $machine->succeed("echo hello");
-      #$machine->waitForUnit('getty@tty2');
-      #$machine->waitForUnit("rogue");
-      $machine->waitForUnit("nixos-manual");
+      machine.succeed("echo hello")
+      # machine.wait_for_unit("getty@tty2")
+      # machine.wait_for_unit("rogue")
+      machine.wait_for_unit("nixos-manual")
 
       # Wait for hard disks to appear in /dev
-      $machine->succeed("udevadm settle");
+      machine.succeed("udevadm settle")
 
       # Partition the disk.
       ${createPartitions}
 
       # Create the NixOS configuration.
-      $machine->succeed("nixos-generate-config --root /mnt");
+      machine.succeed("nixos-generate-config --root /mnt")
 
-      $machine->succeed("cat /mnt/etc/nixos/hardware-configuration.nix >&2");
+      machine.succeed("cat /mnt/etc/nixos/hardware-configuration.nix >&2")
 
-      $machine->copyFileFromHost(
+      machine.copy_file_from_host(
           "${ makeConfig { inherit bootLoader grubVersion grubDevice grubIdentifier grubUseEfi extraConfig; } }",
-          "/mnt/etc/nixos/configuration.nix");
+          "/mnt/etc/nixos/configuration.nix",
+      )
 
       # Perform the installation.
-      $machine->succeed("nixos-install < /dev/null >&2");
+      machine.succeed("nixos-install < /dev/null >&2")
 
       # Do it again to make sure it's idempotent.
-      $machine->succeed("nixos-install < /dev/null >&2");
+      machine.succeed("nixos-install < /dev/null >&2")
 
-      $machine->succeed("umount /mnt/boot || true");
-      $machine->succeed("umount /mnt");
-      $machine->succeed("sync");
+      machine.succeed("umount /mnt/boot || true")
+      machine.succeed("umount /mnt")
+      machine.succeed("sync")
 
-      $machine->shutdown;
+      machine.shutdown()
 
       # Now see if we can boot the installation.
-      $machine = createMachine({ ${hdFlags} qemuFlags => "${qemuFlags}", name => "boot-after-install" });
+      machine = create_machine_from_file(
+          "${ makeMachineConfig "boot-after-install" }"
+      )
 
       # For example to enter LUKS passphrase.
       ${preBootCommands}
 
       # Did /boot get mounted?
-      $machine->waitForUnit("local-fs.target");
+      machine.wait_for_unit("local-fs.target")
 
       ${if bootLoader == "grub" then
-          ''$machine->succeed("test -e /boot/grub");''
+          ''machine.succeed("test -e /boot/grub")''
         else
-          ''$machine->succeed("test -e /boot/loader/loader.conf");''
+          ''machine.succeed("test -e /boot/loader/loader.conf")''
       }
 
       # Check whether /root has correct permissions.
-      $machine->succeed("stat -c '%a' /root") =~ /700/ or die;
+      machine.succeed("stat -c '%a' /root | grep 700")
```

Comment:
why not `assert "700" in machine.succeed(...)`

Reply:
might be possible. It's just done this way everywhere else. If it would be preferable to use `assert` i am fine with it.

Reply:
Yeah, let's use `assert` where possible.